internal/virtualcircuit/update.go

```diff
@@ -0,0 +1,150 @@
+		md5         string
+		tags        []string
+		peerAsn     int
+	)
+	// updateConnectionCmd represents the updateConnectionCmd command
+	updateVirtualCircuitCmd := &cobra.Command{
+		Use:   `update -i <id> [-v <vlan UUID>] [-d <description>] [-n <name>] [-s <speed>] [-t <tags>]`,
+		Short: "Updates a virtualcircuit.",
+		Long:  "Updates a specified virtualcircuit.",
+		Example: `  # Updates a specified virtualcircuit:
+
+	metal vc update [-i <id>] [-n <name>] [-d <description>] [-v <vnid> ] [-s <speed> ] [-t <tags> ]
+
+	metal vc update -i e2edb90b-a8ef-47cb-a577-63b0ba129c29 -d "test-inter-fri-dedicated"
+
+	metal vc update [-i <id>] [-n <name>] [-d <description>] [-M <md5sum>] [-a <peer-asn>] [-S <subnet>] [-c <customer-ip>] [-m <metal-ip>] [-t <tags> ]`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			inc := []string{}
+			var (
+				virtualCircuitUpdateInput metal.VirtualCircuitUpdateInput
+				err                       error
+			)
+			header := []string{"ID", "Name", "Type", "Speed", "Created"}
+
+			vnidFlag, _ := cmd.Flags().GetString("vnid")
+			if vnidFlag != "" {
+				vlanUpdateInput := metal.NewVlanVirtualCircuitUpdateInput()
+				updateParams(vlanUpdateInput, name, description, tags)
+
+				virtualCircuitUpdateInput = updateVlanVirtualCircuit(vlanUpdateInput, vnid, speed)
+			} else {
+				vrfUpdateInput := metal.NewVrfVirtualCircuitUpdateInput()
+
+				updateParams(vrfUpdateInput, name, description, tags)
+
+				virtualCircuitUpdateInput = updateVrfVirtualCircuit(vrfUpdateInput, customerIP, metalIP, subnet, md5, speed, peerAsn)
+			}
+			vc, _, err := c.Service.UpdateVirtualCircuit(context.Background(), vcID).VirtualCircuitUpdateInput(virtualCircuitUpdateInput).Include(c.Servicer.Includes(inc)).Execute()
+			if err != nil {
+				return fmt.Errorf("could not update virtual-circuit: %w", err)
+			}
+
+			data := make([][]string, 1)
+			if vnidFlag != "" {
+				data[0] = []string{vc.VlanVirtualCircuit.GetId(), vc.VlanVirtualCircuit.GetName(), string(vc.VlanVirtualCircuit.GetType()), strconv.Itoa(int(vc.VlanVirtualCircuit.GetSpeed())), vc.VlanVirtualCircuit.GetCreatedAt().String()}
+			} else {
+				data[0] = []string{vc.VrfVirtualCircuit.GetId(), vc.VrfVirtualCircuit.GetName(), string(vc.VrfVirtualCircuit.GetType()), strconv.Itoa(int(vc.VrfVirtualCircuit.GetSpeed())), vc.VrfVirtualCircuit.GetCreatedAt().String()}
+			}
+
+			return c.Out.Output(vc, header, &data)
+		},
+	}
+
+	// Virtual Circuit Params
+	updateVirtualCircuitCmd.Flags().StringVarP(&vcID, "id", "i", "", "Specify the UUID of the virtual-circuit.")
+
+	// Common params for both VlanVirtualCircuit and VrfVirtualCircit
+	updateVirtualCircuitCmd.Flags().StringVarP(&speed, "speed", "s", "", "Adds or updates Speed can be changed only if it is an interconnection on a Dedicated Port")
+	updateVirtualCircuitCmd.Flags().StringSliceVarP(&tags, "tags", "t", []string{}, `updates the tags for the connection --tags="tag1,tag2".`)
```

Comment:
I see that the check for `len(tags) > 0` was removed, but the docs should also be updated to describe the parameter behavior more specifically.  When I pass in tags, does it overwrite existing tags or does it append the tags to the existing ones.   If I run `metal vc update -i <id> -t` with no value given for the `-t` flag, does it remove existing tags?  It would also be good to have at least one test for the update subcommand (I didn't see one but maybe I missed it).

Reply:
CLI users will be looking at the help text, not sample command output.  Could you update the docs for the `--tags` flags so that they accurately reflect the parameter behavior? The flag docs should show any supported formats and the way to remove tags.